src/sentry/sentry_metrics/querying/eap/mql_eap_bridge.py

```diff
@@ -0,0 +1,131 @@
+from collections.abc import Iterable, Sequence
+from datetime import datetime, timedelta
+
+import requests
+import snuba_sdk.mql.mql
+from django.conf import settings
+from google.protobuf.timestamp_pb2 import Timestamp as ProtobufTimestamp
+from snuba_sdk import Timeseries
+from snuba_sdk.conditions import And as MQLAnd
+from snuba_sdk.conditions import Condition as MQLCondition
+from snuba_sdk.conditions import ConditionGroup
+from snuba_sdk.conditions import Op as MQLOp
+from snuba_sdk.conditions import Or as MQLOr
+
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.sentry_metrics.querying.eap.protobufs.AggregateBucket_pb2 import (
+    AggregateBucketRequest,
+    AggregateBucketResponse,
+)
+from sentry.sentry_metrics.querying.eap.protobufs.BaseRequest_pb2 import RequestInfo
+from sentry.sentry_metrics.querying.eap.protobufs.Filters_pb2 import (
+    AndFilter,
+    OrFilter,
+    StringFilter,
+    TraceItemFilter,
+)
+
+
+def parse_mql_filters(group: ConditionGroup) -> Iterable[TraceItemFilter]:
+    for cond in group:
+        if isinstance(cond, MQLAnd):
+            yield TraceItemFilter(
+                and_filter=AndFilter(filters=list(parse_mql_filters(cond.conditions)))
+            )
+        elif isinstance(cond, MQLOr):
+            yield TraceItemFilter(
+                and_filter=OrFilter(filters=list(parse_mql_filters(cond.conditions)))
+            )
+        elif isinstance(cond, MQLCondition):
+            if cond.op == MQLOp.EQ:
+                yield TraceItemFilter(
+                    string_comparison=StringFilter(key=cond.lhs.name, value=cond.rhs)
+                )
+        # TODO: maybe we want to implement other stuff
+
+
+def make_eap_request(
```

Comment:
I noticed that there are a few features of MQL queries that this code doesn't work for:
- Formula aggregates
- Group By clauses
- With totals flag
- Order by

Looks like you are catching the first one already, but it would be good to check for the other and revert back to the original query stream in those cases.

I think this check will catch the first one, but not the other ones

Reply:
this is only for net-new metrics in the virtual "eap" namespace, falling through to the old logic wouldn't work because there wouldn't be any data for them